### DISCUSSION TITLE
docker restart command issue

### DISCUSSION THREAD

**Discussion by RayKimEX** [thijsvanloef/palworld-server-docker#570] [Q&A]
Can you explain why the server gets reset even though I only executed the 'docker restart' command? It is a Linux server, and the save files were located in '/home/[username]/palworld/Pal/Saved/SaveGames/'. However, after executing the 'docker restart' command, those save files are not applied, and it starts from the beginning. How can I deal with this situation, and is there a way to recover it?

**Comment by thijsvanloef** [comment]
Could you share the docker-compose or start command you use? I suspect that your volumes are not persistent

**Reply by RayKimEX** [reply]
The first time I started the server, I used `docker run -d \
    --name palworld-server \
    -p 8211:8211/udp \
    -p 27015:27015/udp \
    -v ./palworld:/palworld/ \
    -e PUID=1000 \
    -e PGID=1000 \
    -e PORT=8211 \
    -e PLAYERS=16 \
    -e MULTITHREADING=true \
    -e RCON_ENABLED=true \
    -e RCON_PORT=25575 \
    -e TZ=UTC \
    -e EXP_RATE=10.000000 \
    -e PAL_CAPTURE_RATE=2.000000 \
    -e PAL_DAMAGE_RATE_ATTACK=0.900000 \
    -e PLAYER_STAMINA_DECREASE_RATE=0.800000 \
    -e PLAYER_AUTO_HP_REGEN_RATE=4.000000 \
    -e PLAYER_AUTO_HP_REGEN_RATE_IN_SLEEP=4.000000 \
    -e PAL_AUTO_HP_REGEN_RATE=4.000000 \
    -e PAL_AUTO_HP_REGEN_RATE_IN_SLEEP=4.000000 \
    -e COLLECTION_DROP_RATE=2.000000 \
    -e COLLECTION_OBJECT_HP_RATE=0.500000 \
    -e ENEMY_DROP_ITEM_RATE=2.000000 \
    -e ENABLE_PLAYER_TO_PLAYER_DAMAGE=True \
    -e ENABLE_INVADER_ENEMY=False \
    -e AUTO_RESET_GUILD_TIME_NO_ONLINE_PLAYERS=1.000000 \
    -e PAL_EGG_DEFAULT_HATCHING_TIME=1.000000 \
    -e WORK_SPEED_RATE=4.000000 \
    -e IS_PVP=True \
    -e CAN_PICKUP_OTHER_GUILD_DEATH_PENALTY_DROP=True \
    -e IS_START_LOCATION_SELECT_BY_MAP=False \
    -e ENABLE_DEFENSE_OTHER_GUILD_PLAYER=True \
    -e ADMIN_PASSWORD="adminpassword" \
    -e SERVER_PASSWORD="serverpassword" \
    -e COMMUNITY=false \
    -e SERVER_NAME="servername" \
    -e SERVER_DESCRIPTION="serverdescription" \
    -e ALLOW_CONNECT_PLATFORM="Steam" \
    --restart unless-stopped \
    --stop-timeout 30 \
    thijsvanloef/palworld-server-docker:latest`


Then, I used just `docker restart [CONTAINER ID] ` to restart the server. As a result, the server got reset.
Did I miss something?

**Reply by RayKimEX** [reply]
I would appreciate your response. plz help

**Reply by MusclePr** [reply]
I tried it but it didn't reproduce.
```shell
$ docker restart palworld-server
$ docker logs palworld-server
```
Does it look like the following?
```log
[2025-02-06 00:05:33] [LOG] RCON executed the command. ShowPlayers
[2025-02-06 00:05:38] [LOG] RCON executed the command. ShowPlayers
[2025-02-06 00:05:44] [LOG] RCON executed the command. ShowPlayers
[2025-02-06 00:05:49] [LOG] RCON executed the command. ShowPlayers
[2025-02-06 00:05:54] [LOG] RCON executed the command. ShowPlayers
[2025-02-06 00:05:59] [LOG] RCON executed the command. ShowPlayers
[2025-02-06 00:06:01] [LOG] RCON executed the command. save
Complete Save
[2025-02-06 00:06:01] [LOG] RCON executed the command. Shutdown 1
The server will shut down in 1 seconds. Please prepare to exit the game.
cli: execute: auth: rcon: dial tcp 127.0.0.1:25575: connect: connection refused
Shutdown handler: cleanup.
****EXECUTING USERMOD****
usermod: no changes
Downloading server files with SteamCMD
Redirecting stderr to '/home/steam/Steam/logs/stderr.txt'
Logging directory: '/home/steam/Steam/logs'
[  0%] Checking for available updates...
[----] Verifying installation...
UpdateUI: skip show logoSteam Console Client (c) Valve Corporation - version 1738027521
-- type 'quit' to exit --
Loading Steam API...OK
"@sSteamCmdForcePlatformType" = "linux"
"@sSteamCmdForcePlatformBitness" = "64"

Connecting anonymously to Steam Public...OK
Waiting for client config...OK
```
